deploy-jenkins.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/bash
+set -eou pipefail
+# Script to run ansible tasks. Use the Makefile instead of running this directly.
+
+${DM_CREDENTIALS_REPO}/sops-wrapper -v > /dev/null
+
+JENKINS_VARS_FILE=$(mktemp)
+PRIVATE_KEY_FILE=$(mktemp)
+trap 'rm $JENKINS_VARS_FILE $PRIVATE_KEY_FILE' EXIT
+
+for varfile in ${DM_CREDENTIALS_REPO}/jenkins-vars/*.yaml
+do
+  ${DM_CREDENTIALS_REPO}/sops-wrapper -d $varfile >> $JENKINS_VARS_FILE
+done
+${DM_CREDENTIALS_REPO}/sops-wrapper -d ${DM_CREDENTIALS_REPO}/aws-keys/ci.pem.enc > $PRIVATE_KEY_FILE
+
+echo ${VIRTUALENV_ROOT}/bin/ansible-playbook \
+  -i playbooks/hosts playbooks/jenkins_playbook.yml \
+  -e @$JENKINS_VARS_FILE \
+  -e "jenkins_public_key='$(ssh-keygen -y -f $PRIVATE_KEY_FILE)'" \
+  --key-file=$PRIVATE_KEY_FILE \
+  -e "dm_credentials_repo=${DM_CREDENTIALS_REPO}" \
+  --tags "${TAGS}" ${EXTRA_VARS:-}
```

Comment:
Why is the `${parameter:-word}` expansion required here?

Reply:
I've used the [bash 'strict mode'](http://redsymbol.net/articles/unofficial-bash-strict-mode/).

`set -u` means that undefined variables become errors. `EXTRA_VARS` is the one variable where it's OK for it to be undefined. So I added that expansion so that it wouldn't get picked up by `set -u`

Reply:
Ooh I didn't know about `-u`, thanks!